processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -132,15 +131,25 @@ public void updateSearchResultSet(
         final Object2IntRBTreeMap<SearchHit> set
     )
     {
-      if (selector != null && !(selector instanceof NullDimensionSelector)) {
-        final IndexedInts vals = selector.getRow();
-        for (int i = 0; i < vals.size(); ++i) {
-          final String dimVal = selector.lookupName(vals.get(i));
-          if (searchQuerySpec.accept(dimVal)) {
-            set.addTo(new SearchHit(outputName, Strings.nullToEmpty(dimVal)), 1);
-            if (set.size() >= limit) {
-              return;
-            }
+      if (selector == null) {
+        // Column doesn't exist
+        return;
+      }
+
+      if (selector.nameLookupPossibleInAdvance()
+          && selector.getValueCardinality() == 1
```

Comment:
Please extract this into an utility function "isNullSelector"

Reply:
Changed.